### DISCUSSION TITLE
ESP devices without PSRAM

### DISCUSSION THREAD

**Discussion by Fabrizz** [HomeSpan/HomeSpan#589] [Q&A] [answered]
Hi, I'm building some devices using HomeSpan, as they work great I made some small pcbs for the devices, now the question, does HomeSpan work on esp devices without PSRAM?

For example the **ESP32-S2-SOLO-N4** (N4 = _4MB flash, no PSRAM_) [Expressif ESP32 S2 Datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-s2-solo-2_esp32-s2-solo-2u_datasheet_en.pdf).
Its pin compatible with other ESPs, so its not that big of a problem, but still.

**Comment by HomeSpan** [comment, marked as answer]
HomeSpan does not use PSRAM and should fine without it.  In fact, most baseline ESP32 chips, as well as the -C3, -S2, and -S3, don't include PSRAM at all (I use all of these for my test scripts).  

If you have a chip with PSRAM, it will only be used to the extent that you create your own PSRAM-based arrays in your sketch.

**Reply by Fabrizz** [reply]
Thanks!
